### ISSUE TITLE
Exception calling "getJobHistoryList" with "3" argument(s): ""

### ISSUE THREAD

**Issue by Maurice-De** [RPWimmer/zabbix-templates-arcserve-udp#7]
Hi,
When I try to run the script I get the following error, I get the error on multiple Arcserve UDP installs, for example a UDP VM and a UDP 8400 appliance:

```
Exception calling "getJobHistoryList" with "3" argument(s): ""
At C:\Program Files\Zabbix Agent\Scripts\Arcserve_UDP_VMCount.ps1:79 char:1
+ $jobHistory = $agent.getJobHistoryList($RecentJobID, $null, $null)
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : SoapException
```

Using the command:
`.\Arcserve_UDP_VMCount.ps1 -user administrator -hostname localhost -pass *** -historycountfrom JobSuccesCount`

I cannot seem to find where it is going wrong, what information is further needed for troubleshooting?

Regards,
Maurice

**Comment by RPWimmer** [comment]
First check the follwing point if you can open the URLs in browser for the connection to the Arcserve UDP appliance:
http://localhost:8015/services/UDPService?wsdl
or
https://localhost:8015/services/UDPService?wsdl

regards
RPW

**Comment by Maurice-De** [comment]
Hi,
The HTTPS link is giving me a long (I think XML) list, so that looks like it is working.

Regards, 
Maurice

**Comment by equipetecnologia** [comment]
Anyone got this resolved?!

**Comment by RPWimmer** [comment]
Hi,
I found out that PowerShell code currently has problems when entering passwords and if the password contains a $ character. The password is then terminated at the $ character. When entering via the CMD line, you can put the password in single quotation marks '.
I haven't managed to patch this in the PowerShell code yet.

regards,
RPW

**Comment by Maurice-De** [comment]
Without special characters in the password I still get the same error. Also when I run it in cmd it will break on the same point, it looks like the login is succesfull, but the information gathering breaks or something.
powershell -file .\Arcserve_UDP_VMCount.ps1 -user administrator -hostname localhost -pass *** -historycountfrom JobSuccesCount

When I use quotes on the -pass it cannot even login and throws the invalid credentials message:
powershell -file .\Arcserve_UDP_VMCount.ps1 -user administrator -hostname localhost -pass '***' -historycountfrom JobSuccesCount

Agent: WebServiceProxy.UDPService
agent.login(<username>,<password>,<hostname>)
Created new session
Exception calling "getJobHistoryList" with "3" argument(s): ""
At C:\Program Files\Zabbix Agent\Scripts\Arcserve.ps1:79 char:1
+ $jobHistory = $agent.getJobHistoryList($RecentJobID, $null, $null)
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : SoapException


Name                           Value
----                           -----
JobMissedCount                 0
JobInProgressCount             0
JobIncompleteCount             0
JobCancelledCount              0
JobSuccessCount                0
JobIdleCount                   0
Timestamp                      1707217108
JobOtherCount                  0
JobWaitingCount                0
JobFailedCount                 0

Maybe it has something to do with powershell versions or something, did you install anything specific for this script? Like the new terminal? I am running Arcserve appliance with Windows Server 2016.

**Comment by equipetecnologia** [comment]
Yes, I m getting same message with Windows Server 2019, even login working fine.

Exception calling "getJobHistoryList" with "3" argument(s): ""
At C:\Program Files\Zabbix Agent\Scripts\Arcserve.ps1:79 char:1
$jobHistory = $agent.getJobHistoryList($RecentJobID, $null, $null)
